src/ExpenseTracker/Models/Account.cs

```diff
@@ -46,8 +47,69 @@ public void AddExpense(decimal expenseAmount, string category)
             Amount = expenseAmount,
             Category = category,
         };
+        this.TotalTransactionDataList.Add(newExpense);
+
         this.CurrentBalance -= expenseAmount;
         this.TotalExpense += expenseAmount;
-        this.TotalTransactionDataList.Add(newExpense);
+    }
+
+    /// <inheritdoc/>
+    public void EditTransactionAmount(int index, decimal newAmountValue)
+    {
+        ITransaction transaction = this.TotalTransactionDataList[index];
+        if (transaction is IncomeTransactionData income)
+        {
+            this.TotalIncome -= income.Amount;
+            this.TotalIncome += newAmountValue;
+            this.CurrentBalance -= income.Amount;
+            this.CurrentBalance += newAmountValue;
+            income.Amount = newAmountValue;
+        }
+        else if (transaction is ExpenseTransactionData expense)
+        {
+            this.TotalExpense -= expense.Amount;
+            this.TotalExpense += newAmountValue;
+            this.CurrentBalance += expense.Amount;
+            this.CurrentBalance -= newAmountValue;
+            expense.Amount = newAmountValue;
+        }
```

Comment:
```suggestion
ITransaction transaction = this.TotalTransactionDataList[index];
this.TotalIncome -= transaction.Amount;
this.TotalIncome += newAmountValue;

if (transaction is IncomeTransactionData income)
{
    this.CurrentBalance -= income.Amount;
    this.CurrentBalance += newAmountValue;
}
else if (transaction is ExpenseTransactionData expense)
{
    this.CurrentBalance += expense.Amount;
    this.CurrentBalance -= newAmountValue;
}

transaction.Amount = newAmountValue;
```

Optional. Understand how the number lines are reduced.